kythe/cxx/indexer/cxx/GraphObserver.h

```diff
@@ -72,6 +70,25 @@ class ProfileBlock {
   const char* SectionName;
 };
 
+class HashRecorder {
```

Comment:
Any particularly reason to go with a single-function interface over just `absl::AnyInvocable` or `std::function`?

Reply:
I find it easier to reason about the lifetime and of the internal state of the class.